### PR TITLE
fix: Refactor dynamic bindings in widgets during building block paste.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
* This pull request refactors the function `handleButtonDynamicTriggerPathList` to `handleWidgetDynamicTriggerPathList` in order to make it more generic and handle dynamic trigger paths for all types of widgets. 
* Additionally, the pull request moves the function `handleIfParentIsListWidgetWhilePasting` out of `handleSpecificCasesWhilePasting` to handle compound cases where the parent of the current widget needs to be checked. 
* Improves the nested object value replacement logic in `PasteWidgetUtils`.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Templates, @tag.Widget, @tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9656587968>
> Commit: 4b821787d7ab5ad302b458c726c632b1e43b49e4
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9656587968&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Templates, @tag.Widget, @tag.Sanity`

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Refactor**
  - Simplified handling of pasting widgets by consolidating functions.
  - Improved logic for handling widget pasting scenarios.
  - Reorganized function names for better clarity and consistency.

- **Bug Fixes**
  - Enhanced the handling of widgets when their parent is a list widget during pasting.

- **Tests**
  - Updated test cases to reflect changes in widget pasting logic and function names.

### CODE REVIEW

**Comment by rahulbarwal** [comment]
@coderabbitai review

File: app/client/src/sagas/BuildingBlockSagas/BuildingBlockAdditionSagas.ts [hunk 1]
```diff
@@ -829,27 +829,24 @@ function handleOtherWidgetReferencesWhilePastingBuildingBlockWidget(
...
+    handleWidgetDynamicTriggerPathList(widgetNameMap, widget);
   }
+  if (!!widget.dynamicBindingPathList) {
+    handleWidgetDynamicBindingPathList(widgetNameMap, widget);
+  }
+  if (!!widget.dynamicPropertyPathList) {
+    handleWidgetDynamicPropertyPathList(widgetNameMap, widget);
+  }
+  widgets = handleIfParentIsListWidgetWhilePasting(widget, widgets);
+
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/sagas/BuildingBlockSagas/BuildingBlockAdditionSagas.ts]: Done.

**Review by jacquesikot — APPROVED** [review]
I changes typo in comment in line 1550 of `WidgetOperationSagas.tsx`

**Comment by riodeuno** [comment]
@rahulbarwal I'm lacking enough context to review this PR. I'm not sure what `handleButtonDynamicTriggerPathList` does. 

In any case, this PR seems to be a refactor for generalisation of this function and doesn't effect the overall system design of the feature.

Such being the case, I'm removing myself as a review from this PR. Please feel free to add me again, if my review is needed.

**Pull request merged**